### DISCUSSION TITLE
I would suggest adding a markdown file showing the code structure, which doesn't make sense to me.

### DISCUSSION THREAD

**Discussion by Bee4Queer** [nasa/fprime#1900] [Ideas]
I really don't understand it

**Comment by thomas-bc** [comment]
There is https://nasa.github.io/fprime/UsersGuide/dev/source-tree.html which goes over the main top level folders of the source tree and what's in them (e.g. the main components under `Fw/` `Os/` etc...). I have also found that a good way of getting a sense of what everything is and how it ties together is to follow the [tutorials](https://nasa.github.io/fprime/Tutorials/README.html).

**Comment by Bee4Queer** [comment]
Ok, thanks.
